astropy/coordinates/sky_coordinate.py

```diff
@@ -1003,6 +1007,105 @@ def to_string(self, style='decimal', **kwargs):
 
         return coord_string
 
+    def to_table(self, table=None, column_conflicts='warn',
+                 metadata_conflicts='warn'):
+        """
+        Convert the data in this |SkyCoord| to a |QTable|.
+
+        Any attributes that have the same length as the |SkyCoord| will be
+        columns in the output. All other attributes will be recorded as
+        metadata.
+
+        If ``table`` is a |Table| whose length matches that of the |SkyCoord|
+        then the columns and metadata of the converted |SkyCoord| are merged
+        with it. In case of a column name or metadata conflict data from the
+        |SkyCoord| is preferred.
+
+        Parameters
+        ----------
+        table : |Table|, optional
+            A table to which the data from the |SkyCoord| is added.
+        column_conflicts : str ('warn'|'error'|'silent')
+            How to handle column name conflicts. Has no effect if ``table`` is
+            not provided.
+        metadata_conflicts : str ('warn'|'error'|'silent')
+            How to handle metadata conflicts. Has no effect if ``table`` is not
+            provided.
+
+        Returns
+        -------
+        |QTable|
+            A |QTable| containing the data of this |SkyCoord|, merged with
+            ``table`` if it is provided.
+
+        Examples
+        --------
+        >>> sc = SkyCoord(ra=[40, 70]*u.deg, dec=[0, -20]*u.deg,
+        ...               obstime=Time([2000, 2010], format='jyear'))
+        >>> t =  sc.to_table()
+        >>> t
+        <QTable length=2>
+           ra     dec   obstime
+          deg     deg
+        float64 float64   Time
+        ------- ------- -------
+           40.0     0.0  2000.0
+           70.0   -20.0  2010.0
+        >>> t.meta
+        {'representation_type': 'spherical', 'frame': 'icrs'}
+        >>> t = Table({'a': [3, 7]}, meta={'b': 'c'})
+        >>> t = sc.to_table(t)
+        >>> t
+        <QTable length=2>
+          a      ra     dec   obstime
+                deg     deg
+        int64 float64 float64   Time
+        ----- ------- ------- -------
+            3    40.0     0.0  2000.0
+            7    70.0   -20.0  2010.0
+        >>> t.meta
+        {'b': 'c', 'representation_type': 'spherical', 'frame': 'icrs'}
+        """
+        self_as_dict = self.info._represent_as_dict()
+        tabledata = {}
+        metadata = {}
+        # Record attributes that have the same length as self as columns in the
+        # table, and the other attributes as table metadata.  This matches
+        # table.serialize._represent_mixin_as_column().
```

Comment:
It would be nice to have more comments in this block of code.
What is each if-block doing?

Reply:
I would say that at least for the blocks between lines 1083 and 1091 the error messages already provide adequate explanations.